src/models/blockchain/block/validity/block_primitive_witness.rs

```diff
@@ -225,6 +225,10 @@ pub(crate) mod test {
     }
 
     impl BlockPrimitiveWitness {
+        pub(crate) fn predecessor_block(&self) -> Block {
```

Comment:
cloning a block (to_owned) is expensive.    Why not return &Block.   Caller can then choose whether it needs to clone it or not.

Reply:
Sure. This is test-code, but I can do that.